### DISCUSSION TITLE
byte[]  vs. byte?

### DISCUSSION THREAD

**Discussion by preinagel** [bonsai-rx/bonsai#1556] [Q&A] [answered]
What does the notation [] mean?

The output of a SerialRead node is type byte[]
I want to condition something on the value of that byte  
But the Equal operator cannot accept a  byte[] as an input, it wants type byte
How do I convert a byte[] into a byte?

**Comment by bruno-f-cruz** [comment, marked as answer]
`[]` stands for array. You can find more [information here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/arrays).

The output of SerialRead will be an array of bytes. The size depends on the `Count` property.
If you want to index single bytes of the array, you can use `Index`. If you want to "flatten" all values of the array to an event sequence of individual bytes, you can use the `Merge(rx)` operator.
Hope it helps

**Reply by preinagel** [reply]
Thanks.
